### PR TITLE
feat: add HD-EPIC VQA benchmark (CVPR 2025)

### PR DESCRIPTION
**Pull request by aliazani**
## Summary
- Adds HD-EPIC VQA benchmark (Perrett et al., CVPR 2025), a 26,550-question egocentric kitchen video QA benchmark across 30 prototypes and 7 categories
- Provides per-prototype, per-category, and full benchmark task configs runnable via `--tasks hd_epic_<prototype>`, `--tasks hd_epic_<category>`, or `--tasks hd_epic`
- Validated against published zero-shot Qwen2.5-VL-7B baseline from the HD-EPIC challenge community report

## In scope
- 30 per-prototype YAML task configs + 7 category group YAMLs + 1 master group YAML (`hd_epic`)
- `utils.py` with `doc_to_messages`, `doc_to_visual`, `doc_to_text`, `process_results`, `doc_to_target`, and `<TIME>`/`<BBOX>` tag resolution
- `hd_epic_to_hf.py` converter from official annotation JSONs to lmms-eval JSONL format
- `generate_task_yamls.py` script for regenerating all YAMLs
- `README.md` with setup instructions and validation results

## Out of scope
- Hosting videos or JSONL on HuggingFace (videos are 41 hours; JSONL is generated locally from official annotations)
- Fine-tuned model evaluation
- Changes to any existing tasks or models

## Validation
- `python -m lmms_eval --model qwen2_5_vl --model_args pretrained=Qwen/Qwen2.5-VL-7B-Instruct,fps=1,max_num_frames=32,min_pixels=50176,max_pixels=50176 --tasks hd_epic_ingredient_ingredient_weight --batch_size 1` | sample size: `N=50` | key metrics: `accuracy` | result: **pass** (26%, R3 community report baseline: ~28%, within SE of 6pp)

## Risk / Compatibility
- No changes to existing tasks, models, or shared infrastructure
- Requires `ffmpeg` in PATH and `HD_EPIC_VIDEO_DIR` env var set at eval time; gracefully falls back to full video if clip extraction fails

## Type of Change
- [x] New benchmark/task

### CODE REVIEW

**Review by kcz358 — COMMENTED** [review]
Hi, the tasks LGTM. It would be better if you can split the yaml into sub folders so that the categories are clearer for the user. Thanks

**Comment by aliazani** [comment]
> Hi, the tasks LGTM. It would be better if you can split the yaml into sub folders so that the categories are clearer for the user. Thanks

@kcz358 
Done — pushed in the latest commits. Each of the 7 HD-EPIC categories now has its own subfolder under `tasks/hd_epic/` containing its per-prototype YAMLs and category group YAML. `_hd_epic_base.yaml` and the master group YAML stay at the top level. `generate_task_yamls.py` has been updated to emit this layout going forward.

One thing worth noting: lmms-eval resolves `!function` module names relative to each YAML's own directory, so moving the YAMLs one level deeper broke `!function utils.filter_*` lookups. Fixed by adding a small `utils.py` shim to each category subfolder that re-exports the shared helpers from the top-level `utils.py` — no YAML or top-level code changes required.

Also took the opportunity to add a "Repository layout" section to the README documenting the new structure, and fixed a stale note in the Notes section (clip extraction now uses two-pass ffmpeg seek rather than `-c copy`).

Task names and `--tasks` invocations are unchanged throughout. Let me know if anything needs adjusting!

Original Benchmark link: https://github.com/hd-epic/hd-epic-vqa-eval/tree/main

**Comment by kcz358** [comment]
Thanks for the revision, looks clean and good to merge.

**Comment by aliazani** [comment]
> Thanks for the revision, looks clean and good to merge.

Thanks for merging! I realized I missed reformatting a few files. I’ll send a quick follow-up PR with formatting-only changes.
